docs/yul.rst

```diff
@@ -506,6 +507,8 @@ In variable declarations and assignments, the right-hand-side expression
 variables on the left-hand-side.
 This is the only situation where an expression evaluating
 to more than one value is allowed.
+The same variable name cannot occur more than once in the left-hand-side of
+an assignment or variable declaration.
```

Comment:
There's probably no need to mention variable declarations here, because that's "caught" by the scoping rules, but I haven't actually checked that that's the case and mentioning it here doesn't hurt I guess.